nerscspawner/nerscspawner.py

```diff
@@ -104,24 +105,26 @@ def clear_state(self):
         self.child_profile = ""
 
 
-class NERSCSlurmSpawner(BatchSpawnerRegexStates):
+class NERSCSlurmSpawnerBase(BatchSpawnerRegexStates):
     """Spawner that connects to a job-submit (login node) and submits a job to
     start a process running in the Slurm batch queue.
 
     NOTE Right now we allow the hub to pre-select a random port but when multiple
     users are on the same compute node, a la shared-interactive, we need to control
     the port selected deterministically or ensure they don't collide in some way."""
 
-    # all these req_foo traits will be available as substvars for templated strings
+    # Not all inheriting classes need a value for this attribute
+    req_nodelist = None
```

Comment:
This only goes in one child class.

This should be a traitlet, probably a list.